avxjudge.py

```diff
@@ -157,144 +167,87 @@ def summarize(table: dict, is_pct: bool, max_funcs: int = 5) -> None:
-    total_avx512_score = 0.0
-
-
-    out, err = subprocess.Popen(["objdump","-d", filename], stdout=subprocess.PIPE, stderr=subprocess.PIPE).communicate()
-    alllines = out.decode("latin-1")
-    lines =  alllines.split("\n")
-
-    for line in lines:
-        score_sse = -1.0
-        score_avx2 = -1.0
-        score_avx512 = -1.0
-
-        sse_str = " "
-        avx2_str = " "
-        avx512_str = ""
-
-        match = re.search("^(.*)\#.*", line)
-        if match:
-            line = match.group(1)
-        
-        match = re.search(".*[0-9a-f]+\:\t[0-9a-f\ ]+\t([a-zA-Z0-9]+) (.*)", line)
-
-        if match:
-            ins = match.group(1)
-            arg = match.group(2)
-
-            score_sse = is_sse(ins, arg)
-            score_avx2 = is_avx2(ins, arg)
-            score_avx512 = is_avx512(ins, arg)
-
-            avx2_str= " "
-            instructions += 1
-
-        match = re.search("\<([a-zA-Z0-9_@\.\-]+)\>\:", line)
-        if match:
-            funcname = match.group(1)
-            if instructions > 0 and verbose > 0:
-                print(function,"\t",ratio(sse_count/instructions),"\t", ratio(avx2_count / instructions), "\t", ratio(avx512_count/instructions), "\t", avx2_score,"\t", avx512_score)
-
-            if sse_count >= 1:
-                sse_functions[function] = sse_score
-                sse_functions_ratio[function] = 100.0 * sse_count / instructions
-            if avx2_count >= 1:
-                avx2_functions[function] = avx2_score
-                avx2_functions_ratio[function] = 100.0 * avx2_count / instructions
-            if avx512_count >= 1:
-                avx512_functions[function] = avx512_score
-                avx512_functions_ratio[function] = 100.0 * avx512_count/instructions
-
-            total_sse_count += sse_count
-            total_sse_score += sse_score
-
-            total_avx2_count += avx2_count
-            total_avx2_score += avx2_score
-
-            total_avx512_count += avx512_count
-            total_avx512_score += avx512_score
-
-            instructions = 0
-            function = funcname
-
-
-            sse_count = 0
-            avx2_count = 0
-            avx512_count = 0
-
-            sse_score = 0.0
-            avx2_score = 0.0
-            avx512_score = 0.0
-
-        if score_sse >= 0.0:
-            sse_str = str(score_sse)
-            sse_score += score_sse
-            sse_count += 1
-
-        if score_avx2 >= 0.0:
-            avx2_str = str(score_avx2)
-            avx2_score += score_avx2
-            avx2_count += 1
-
-        if score_avx512 >= 0.0:
-            avx512_str = str(score_avx512)
-            avx512_score += score_avx512
-            avx512_count += 1
-
-
-        if verbose > 0:
-            print(sse_str,"\t",avx2_str,"\t", avx512_str,"\t", line)
+    p = subprocess.Popen(["objdump","-d", filename], stdout=subprocess.PIPE)
+    for line in p.stdout:
+        process_objdump_line(records, line.decode("latin-1"), verbose)
```

Comment:
What's the difference between `p.stdout` and the first return of `p.communicate()`?

Reply:
The p.stdout is basically in reading content as it comes in line by line where as p.communicate() will block until the process finishes. The extra read of p.communicate() content is just in case there was some extra bits that didn't get picked up looping through p.stdout. I'm fuzzy on this being strictly needed but I have some memory where this mattered. I should recheck where I got this idea from and see if that code can't be dropped though.